backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -498,6 +499,29 @@ public void appendBehaviorGroupToEventType(final String orgId, final UUID behavi
         }
     }
 
+    @Transactional
+    public void appendActionToBehaviorGroup(final String orgId, final UUID behaviorGroupUuid, final UUID endpointUuid, final int position) {
+        final String appendBehaviorGroupQuery =
+            "INSERT INTO " +
+                "behavior_group_action(behavior_group_id, endpoint_id, created, position) " +
+                "SELECT bg.id, ep.id, :created, :position " +
+                "FROM behavior_group AS bg, endpoints AS ep " +
+                "WHERE ep.id = :endpointUuid AND ep.org_id = :orgId AND bg.id = :behaviorGroupUuid AND bg.org_id = :orgId " +
+                "ON CONFLICT (behavior_group_id, endpoint_id) DO NOTHING";
```

Comment:
Does the position need to be updated in case of conflict?
```suggestion
                "ON CONFLICT (behavior_group_id, endpoint_id) DO UPDATE SET position = :position";
```